syntax = "proto3";
package RAFT;

service Node{
    rpc RequestVote(RequestVoteRequest) returns (RequestVoteResponse);
}

message RequestVoteRequest{
    int32 term = 1;
    int32 candidate_id = 2;
    int32 last_log_index = 3;
    int32 last_log_term = 4;
}
    
message RequestVoteResponse{
    int32 term = 1;
    int32 voter_id = 2;
    bool vote_granted = 3;
}